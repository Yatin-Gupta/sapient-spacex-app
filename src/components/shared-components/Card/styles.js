/**
 * Card Stylesheet in JSS
 */
const cardBackgroundColor = "#ffffff";
const cardShadowColor = "#4c4c4c";

const styles = {
  card: {
    backgroundColor: cardBackgroundColor,
    borderRadius: "5px",
    width: "100%",
    padding: "10px",
    boxShadow: `1px 1px 3px ${cardShadowColor}`,
  },
};

export default styles;
