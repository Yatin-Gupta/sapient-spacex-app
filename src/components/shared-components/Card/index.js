import React from "react";
import { createUseStyles } from "react-jss";
import styles from "./styles";

const useStyles = createUseStyles(styles);

/**
 * Card Component
 * @param {*} props
 */
function Card(props) {
  const classes = useStyles();
  let className = props.className ? `${classes.card} ${props.className}` : classes.card;
  return (
    <React.Fragment>
      <div className={className} data-testid="card">
        {props.children}
      </div>
    </React.Fragment>
  );
}

export default Card;
