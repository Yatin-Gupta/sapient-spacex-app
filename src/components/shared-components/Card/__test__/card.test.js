import React from "react";
import ReactDOM from "react-dom";
import Card from "../index";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Card component renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Card></Card>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Card component renders with className passed as props", () => {
  const { getByTestId } = render(<Card className="h-100"></Card>);
  expect(getByTestId("card")).toHaveClass("h-100");
});

it("Card component matches the snapshot", () => {
  const tree = renderer.create(<Card></Card>).toJSON();
  expect(tree).toMatchSnapshot();
});
