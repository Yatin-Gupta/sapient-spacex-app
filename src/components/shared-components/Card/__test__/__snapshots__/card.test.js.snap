// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches the snapshot 1`] = `
<div
  className="card-0-2-1"
  data-testid="card"
/>
`;
